api/src/channel/lib/Handler.ts

```diff
@@ -311,6 +311,7 @@ export default abstract class ChannelHandler<
     const [name, _suffix] = this.getName().split('-');
     if ('id' in attachment) {
```

Comment:
If attachment is null or undefined, the code will crash before it ever reaches the inner check or the warning log. Shouldn't we check attachment first ?
```suggestion
    if (attachment && 'id' in attachment
```